docs/features/incremental-generators.md

```diff
@@ -42,154 +42,711 @@ public class MyGenerator : IIncrementalGenerator { ... }
 An assembly can contain a mix of diagnostic analyzers, source generators and
 incremental generators.
 
-### Initialization
-
-`IIncrementalGenerator` has an `Initialize` method that is called by the host
-(either the IDE or the command-line compiler) exactly once, regardless of the
-number of further compilations that may occur. `Initialize` passes an instance
-of `IncrementalGeneratorInitializationContext` which can be used by the
-generator to register a set of callbacks that affect how future generation
-passes will occur.
-
-Currently `IncrementalGeneratorInitializationContext` supports two callbacks:
+### Pipeline based execution
 
-- `RegisterForPostInitialization(GeneratorPostInitializationContext)`: the same
-  callback in shape and function as supported by Source Generators today
-- `RegisterExecutionPipeline(IncrementalGeneratorPipelineContext)`: replaces Execute, described below
+`IIncrementalGenerator` has an `Initialize` method that is called by the host[^1]
+exactly once, regardless of the number of further compilations that may occur.
```

Comment:
May want to give an example here. For example discuss how the IDE may re-use the generator across many different projects in the solution or even within projects for several rounds of typing.